Python/pywarpx/WarpX.py

```diff
@@ -101,7 +103,12 @@ def write_inputs(self, filename='inputs', **kw):
 
         with open(filename, 'w') as ff:
 
+            prefix_old = ''
             for arg in argv:
+                prefix_new = re.split(' |\.', arg)[0]
+                if prefix_new != prefix_old:
+                    ff.write(f'# {prefix_new}\n')
```

Comment:
Can you add a comment here explaining what this is for? Something like

# This prints the name of the input group as a header before each group to improve the human readability of the input file.